aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -752,8 +757,13 @@ Tensor &all_out(Tensor &result, const Tensor &self, int64_t dim, bool keepdim) {
   if (_dimreduce_return_trivial(result, self, 1, dim, keepdim)) {
     return result;
   } else {
-    auto iter = make_reduction(
-      "all", result, self, dim, keepdim, self.scalar_type());
+    if (self.is_cuda()) {
+      auto iter = make_reduction(
+          "all", result, self, dim, keepdim, self.scalar_type(), kBool);
```

Comment:
What's the behavior here if the result tensor does not have boolean dtype?

Reply:
Have added an explicit check for result dtype to be `bool` have added a relevant test as well.